src/features/island/buildings/components/building/tent/TentModal.tsx

```diff
@@ -0,0 +1,126 @@
+    },
+  ] = useActor(gameService);
+
+  const [selectedBumpkin, setSelectedBumpkin] = useState<{
+    equipped: BumpkinParts;
+    tokenId: number;
+  }>(interpretTokenUri(bumpkins[defaultSelectedIndex ?? 0].tokenURI));
+
+  if (!bumpkins) return null;
+
+  const placedTents = (state.buildings.Tent || []).length;
+  const allowedBumpkins = placedTents + DEFAULT_BUMPKIN_ALLOWANCE;
+
+  const farmingBumpkins = bumpkins.slice(0, allowedBumpkins);
+  const nonFarmingBumpkins = bumpkins.slice(allowedBumpkins);
+
+  const MainContent = () => (
+    <div className="flex flex-col space-y-4">
+      <div>
+        <p className="text-sm mb-1">Showing on Farm</p>
+        <div className="flex">
+          {farmingBumpkins
+            .map((bumpkin) => interpretTokenUri(bumpkin.tokenURI))
+            .map(({ tokenId, equipped }) => (
+              <BumpkinBox
+                key={tokenId}
+                bumpkin={{ equipped, id: tokenId }}
+                selectedId={selectedBumpkin.tokenId}
+                onSelect={(tokenId: number) =>
+                  setSelectedBumpkin({ tokenId, equipped })
+                }
+              />
+            ))}
+        </div>
+      </div>
+      {nonFarmingBumpkins.length > 0 && (
+        <div>
+          <p className="text-sm mb-1">In Wallet</p>
+          <div className="flex">
+            {nonFarmingBumpkins
+              .map((bumpkin) => interpretTokenUri(bumpkin.tokenURI))
+              .map(({ tokenId, equipped }) => (
+                <BumpkinBox
+                  key={tokenId}
+                  bumpkin={{ equipped, id: tokenId }}
+                  selectedId={selectedBumpkin.tokenId}
+                  onSelect={(tokenId: number) =>
+                    setSelectedBumpkin({ tokenId, equipped })
+                  }
+                />
+              ))}
+          </div>
+        </div>
+      )}
+    </div>
+  );
+
+  const PanelContent = () => (
+    <div className="flex flex-col space-y-2">
+      <div className="flex space-x-2 sm:flex-col sm:space-x-0">
+        <div className="rounded overflow-hidden w-32 sm:w-full relative">
+          <img
+            src={getBumpkinUrl(selectedBumpkin.equipped)}
+            alt="Selected bumpkin"
+            className="w-full"
+          />
+        </div>
+        <div className="flex justify-center w-full my-2 text-sm">
+          <a
+            href={`${baseUrl}/${selectedBumpkin.tokenId}`}
+            className="underline hover:text-blue-500"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            {`Bumpkin #${selectedBumpkin.tokenId}`}
+          </a>
+        </div>
+      </div>
+    </div>
+  );
+
+  return (
+    <CloseButtonPanel
+      tabs={[{ name: "Bumpkins", icon: SUNNYSIDE.icons.player }]}
+      currentTab={0}
+      onClose={onClose}
+    >
+      <SplitScreenView
+        content={MainContent()}
+        panel={PanelContent()}
+        contentScrollable={false}
```

Comment:
Will allow scrolling when overflow occurs.

```suggestion
        contentScrollable
```